hadoop-hdds/docs/content/design/storage-distribution.md

```diff
@@ -0,0 +1,346 @@
+  - SCM maintains and updates a deletion transaction summary for newly created/deleted transactions. Summary is exposed through metrics, API, and also persisted.
+  - Recon gets the deletion transaction summary using SCM client whenever required.
+- SCM → Datanode
+  - Block deletion transaction sent by SCM to DN carries block size.
+- DataNode → Recon
+  - Exposes TotalPendingDeletionBytes via JMX through BlockDeletingService.
+  - Recon fetches this data for node-level pending deletion tracking.
+- Recon
+  - Consolidates all component-level data.
+  - Computes global storage, namespace usage, and pending deletion metrics.
+  - Exposes results via RestFull API /storagedistribution.
+
+### Recon API Design
+
+The Recon API will expose a single endpoint that returns a JSON object with the following structure:
+
+| Field              | Type             | Description                            |
+|--------------------|------------------|----------------------------------------|
+| globalStorage      | Object           | Overall storage capacity metrics.      |
+| globalNamespace    | Object           | Overall namespace (key) usage metrics. |
+| usedSpaceBreakdown | Object           | Detailed breakdown of used space.      |
+| dataNodeUsage      | Array of Objects | Per-DataNode storage usage.            |
+
+#### globalStorage Object
+
+| Field          | Type | Description                                       |
+|----------------|------|---------------------------------------------------|
+| totalUsedSpace | Long | Total used space across all DataNodes in bytes.   |
+| totalFreeSpace | Long | Total free space across all DataNodes in bytes.   |
+| totalCapacity  | Long | Total raw capacity across all DataNodes in bytes. |
+
+#### globalNamespace Object
+
+| Field          | Type | Description                                             |
+|----------------|------|---------------------------------------------------------|
+| totalUsedSpace | Long | Total used space for namespace objects (keys) in bytes. |
+| totalKeys      | Long | Total number of keys in the namespace.                  |
+
+#### usedSpaceBreakdown Object
+
+| Field                 | Type   | Description                                            |
+|-----------------------|--------|--------------------------------------------------------|
+| openKeysBytes         | Long   | Bytes currently held by open keys (not yet committed). |
+| committedBytes        | Long   | Bytes committed to existing keys.                      |
+| containerPreAllocated | Long   | Pre-allocated space for open containers.               |
+| deletionPendingBytes  | Object | Bytes pending deletion.                                |
+
+#### deletionPendingBytes Object
+
+| Field   | Type   | Description                                     |
+|---------|--------|-------------------------------------------------|
+| total   | Long   | Total bytes pending deletion across all stages. |
```

Comment:
I assume this is replicated size since it needs to make sense when aggregated from both OM and datanodes. We should include this in the description of the API documentaiton.